crowdsourcehinter/static/js/src/crowdsourcehinter.js

```diff
@@ -0,0 +1,234 @@
+function CrowdsourceHinter(runtime, element, data){
+    //switching units back to a previous question will make a second hinter start up.
+    //executeHinter is used to disable the hinter after switching units in an edX course.
+    var executeHinter = true;
+    $(".crowdsourcehinter_block", element).hide();
+
+    if(executeHinter){
+    var isShowingHintFeedback = false;
+    var voted = false;
+    $(".csh_HintsToUse", element).text("");
+
+    function stopScript(){
+    //This function is used to prevent a particular instance of the hinter from acting after
+    //switching between edX course's units. 
+        executeHinter = false;
+    }
+    Logger.listen('seq_next', null, stopScript);
+    Logger.listen('seq_prev', null, stopScript);
+    Logger.listen('seq_goto', null, stopScript);
+
+    //data about the problem obtained from Logger.listen('problem_graded') is passed on to the onStudentSubmission.
+    //directly passing data to onStudentSubmission does not appear to work
+    function get_event_data(event_type, data, element){
+        onStudentSubmission(data);
+    }
+    Logger.listen('problem_graded', data.hinting_element, get_event_data);
+
+    function onStudentSubmission(problem_graded_event_data){
+    //This function will determine whether or not the student correctly answered the question.
+    //If it was correctly answered it will begin the process for giving feedback on hints.
+        if (problem_graded_event_data[1].search(/class="correct/) === -1){
+            $(".crowdsourcehinter_block", element).show();
+            $.ajax({
+                type: "POST",
+                url: runtime.handlerUrl(element, 'get_hint'),
+                data: JSON.stringify({"submittedanswer": unescape(problem_graded_event_data[0])}),
+                success: seehint
+            });
+        }else{
+            $('.csh_correct', element).show();
+            $(".csh_hint_reveal", element).hide();
+            //send empty data for ajax call because not having a data field causes error
+            $.ajax({
+                type: "POST",
+                url: runtime.handlerUrl(element, 'get_feedback'),
+                data: JSON.stringify({}),
+                success: getFeedback
+            });
+        }  
+    }
+
+    function seehint(result){
+    //Show a hint to the student after an incorrect answer is submitted.
```

Comment:
See https://github.com/pmitros/RecommenderXBlock/blob/master/recommender/static/js/src/recommender.js

This will give an example of how to write good function docs. 

1. They go in /* */ comments above the function
2. They document inputs/outputs when necessary
3. Docs can be automatically extracted from something like this.